From ad9870aca4e04735755201ce26715076b906275d Mon Sep 17 00:00:00 2001
From: Li Wencheng <liwencheng@phytium.com.cn>
Date: Wed, 7 May 2025 17:22:12 +0800
Subject: [PATCH 36/67] PHYTIUM: net/phytmac: Cancels the power-on/off
 capability

The MAC register is powered on by default in the firmware
and the driver cancels the power-on and power-off capability.

Signed-off-by: Li Wencheng <liwencheng@phytium.com.cn>
Signed-off-by: Wang Yinfeng <wangyinfeng@phytium.com.cn>
Change-Id: I6a55c577f92fe4f39bc7290861f217b7699ba9ee

Link: https://gitee.com/phytium_embedded/phytium-linux-kernel/commit/ba2e5fa1f8e4e92786347336f7fd56c2cd4f65fc
Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>
---
 drivers/net/ethernet/phytium/phytmac.h          | 2 +-
 drivers/net/ethernet/phytium/phytmac_platform.c | 1 -
 2 files changed, 1 insertion(+), 2 deletions(-)

diff --git a/drivers/net/ethernet/phytium/phytmac.h b/drivers/net/ethernet/phytium/phytmac.h
index 1351f8f7a895..c53277e56631 100644
--- a/drivers/net/ethernet/phytium/phytmac.h
+++ b/drivers/net/ethernet/phytium/phytmac.h
@@ -15,7 +15,7 @@

 #define PHYTMAC_DRV_NAME		"phytium-mac"
 #define PHYTMAC_DRV_DESC		"PHYTIUM Ethernet Driver"
-#define PHYTMAC_DRIVER_VERSION		"1.0.41"
+#define PHYTMAC_DRIVER_VERSION		"1.0.42"
 #define PHYTMAC_DEFAULT_MSG_ENABLE	  \
 		(NETIF_MSG_DRV		| \
 		NETIF_MSG_PROBE	| \
diff --git a/drivers/net/ethernet/phytium/phytmac_platform.c b/drivers/net/ethernet/phytium/phytmac_platform.c
index 95509711894b..eb3cc7d8f429 100644
--- a/drivers/net/ethernet/phytium/phytmac_platform.c
+++ b/drivers/net/ethernet/phytium/phytmac_platform.c
@@ -30,7 +30,6 @@ static const struct phytmac_config phytium_2p0_config = {
 	.hw_if = &phytmac_2p0_hw,
 	.caps = PHYTMAC_CAPS_TAILPTR
 			| PHYTMAC_CAPS_RXPTR
-			| PHYTMAC_CAPS_PWCTRL
 			| PHYTMAC_CAPS_LSO
 			| PHYTMAC_CAPS_MSG
 			| PHYTMAC_CAPS_JUMBO,
--
2.50.1
